app/components/footer.tsx

```diff
@@ -5,6 +5,33 @@ import { forwardRef, ReactNode } from "react";
 import contentRoutes from "../content-routes.json";
 import { useLocale } from "../locales/use-locale";
 
+const Version = () => {
+  let commitLink = null;
+
+  if (process.env.NEXT_PUBLIC_GITHUB_REPO && process.env.NEXT_PUBLIC_COMMIT) {
```

Comment:
In which case do we not have have GITHUB_REPO in next public variables ?
 Nit: for readability I think assign the env variables to local variables would be a tidbit better.

```
const githubRepo = process.env.NEXT_PUBLIC_GITHUB_REPO
```

Reply:
Ah maybe in preview deployments ?

Reply:
We always have it, it's actually set in next.config.js and comes from package.json (like `NEXT_PUBLIC_VERSION`)

Reply:
Regarding the nit: we actually have an `env.ts` file which reads and re-exports env variables. Maybe that would be the place to do this? I'm not super happy with that because currently it mixes runtime and build env vars. Since runtime env vars are only available server-side, this doesn't always work as expected.

What do you think about splitting that up into two files, e.g.

```ts
// env-build.ts
export const VERSION = process.env.NEXT_PUBLIC_VERSION;
// etc.

// env-server.ts
export const PUBLIC_URL = process.env.PUBLIC_URL;
// ...
```

Reply:
So it seems like the else if will never be entered right ?

Reply:
I like the idea of making the environment explicit.

Reply:
> So it seems like the else if will never be entered right ?

You mean since GITHUB_REPO is always set we shouldn't have this condition in the first place? Maybe … I think since from the frontend perspective this is still optional, we should handle the case where it's undefined.